### ISSUE TITLE
<date-part range-delimiter="..." /> attribute not honoured at all

### ISSUE THREAD

**Issue by cormacrelf** [Juris-M/citeproc-js#91]
CSL spec: http://docs.citationstyles.org/en/stable/specification.html#date-ranges

Date ranges are supposed to be customizable with the `range-delimiter` attribute. citeproc-js does not implement this.

Reading the source, it appears that the only delimiter used for dates is the `year-range-delimiter` and (sometimes?) the `citation-range-delimiter` (`src/node_datepart.js`). Anyhow, the following date block:

```xml
    <date form="text" delimiter=" ">
        <date-part name="day" form="numeric" range-delimiter="ANYTHING" />
        <date-part name="month" form="long" range-delimiter=" ANYTHING " />
        <date-part name="year" form="numeric" range-delimiter=" ANYTHING "/>
    </date>
```

renders `21–22 July 2018`, `1 July–5August 2018` (which is wrong twice - no space in 5August) and `July–August 2018` etc. Nothing you put in range-delimiter has any effect.

And sure enough,

```xml
      <term name="year-range-delimiter"> YRD </term>
```

... means you get a bunch of YRD in the output.

`pandoc-citeproc` gets this almost right, bar the 5August, but you can use suffixes instead of delimiter=" " to get the right output:

![screen shot 2018-11-25 at 17 37 49](https://user-images.githubusercontent.com/378760/48976293-0955fd80-f0d9-11e8-856c-7a57a50e7180.png)

**Comment by fbennett** [comment]
I see that this has been in the spec for the past nine years, but I have no memory of implementing it. I don't think I ever noticed that it was there. Implementation will take some effort, as the date handling code is pretty messy. Will keep it in the list, though, and hope to get to it eventually.

**Comment by fbennett** [comment]
Fixed now, at [tag 1.1.222](https://github.com/Juris-M/citeproc-js/tree/1.1.222). Thanks for catching this - I had totally walked past it.
